test/app/e2e/relay/server.js

```diff
@@ -0,0 +1,212 @@
+// @ts-check
+
+'use strict';
+
+const ngrok = require('@ngrok/ngrok');
+const axios = require('axios').default;
+const express = require('express');
+const cors = require('cors');
+const twilio = require('twilio');
+
+/** @type {Map<string, any[]>} */
+const receivedMessages = new Map();
+
+/**
+ * @param {string} envVarKey
+ * @returns {string}
+ */
+const getEnvVar = (envVarKey) => {
+  const envVarVal = process.env[envVarKey];
+  if (typeof envVarVal !== 'string' || envVarKey.length === 0) {
+    throw new Error(`"${envVarKey}" env var not defined.`);
+  }
+  return envVarVal;
+};
+
+/**
+ * @returns {{
+ *    apiKeySid: string;
+ *    apiKeySecret: string;
+ *    accountSid: string;
+ *    authToken: string;
+ * }}
+ */
+const getEnvVars = () => {
+  const apiKeySid = getEnvVar('API_KEY_SID');
+  const apiKeySecret = getEnvVar('API_KEY_SECRET');
+  const accountSid = getEnvVar('ACCOUNT_SID');
+  const authToken = getEnvVar('AUTH_TOKEN');
+  return { apiKeySid, apiKeySecret, accountSid, authToken };
+};
+
+/**
+ * @typedef {(
+ *   callSid: string,
+ *   callbackUrl: string,
+ * ) => Promise<import('axios').AxiosResponse>} CreateSubscription
+ */
+
+/**
+ * @typedef {(
+ *   callSid: string,
+ * ) => Promise<import('axios').AxiosResponse>} SendMessage
+ */
+
+/**
+ * @param {string} accountSid
+ * @param {string} apiKeySid
+ * @param {string} apiKeySecret
+ */
+const createTwilioUserDefinedMessageActions = (
+  accountSid,
+  apiKeySid,
+  apiKeySecret,
+) => {
+  const baseUrl = 'https://api.twilio.com/2010-04-01';
+  const extendedUrl = `${baseUrl}/Accounts/${accountSid}/Calls`;
+
+  /**
+   * @type {CreateSubscription}
+   */
+  const createSubscription = (callSid, callbackUrl) => {
+    const url =
+      `${extendedUrl}/${callSid}/UserDefinedMessageSubscriptions.json`;
+
+    const params = new URLSearchParams();
+    params.append('Method', 'POST');
```

Comment:
is this `POST` needed if you already using `post` on line 79?

Reply:
We need it in both locations, yes. The Twilio API defines the protocol as such. The first POST (in params append) is actually part of the data we're sending to Twilio. The second POST is the actual POST request we're making.